src/kernel/filesystem/device.zig

```diff
@@ -0,0 +1,61 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+/// The functions for a hardware device. A device can be a RAM device where the all content is in
+/// RAM, or can be a hard disk drive device for a floppy or CD etc.
+pub const Device = struct {
```

Comment:
I was thinking about the name and wondered if something like `Resource` would be more appropriate to avoid confusion with hardware devices (like mice, graphics cards, controllers etc.)? If we can't think of a better name then `Device` is fine.

Reply:
How about `StorageDevice`

Reply:
Hmm "storage" typically means persistent storage, like disks, flash drives etc.

Reply:
I guess `Resource` is fine